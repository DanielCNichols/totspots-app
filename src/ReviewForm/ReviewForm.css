.review_form_component {
  text-align: center;
  margin: 100px auto;
}


.review_form {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 100px;
}
.review_form label {
  text-align: center;
  display: block;
  margin: 1.5em;
}

.review_form_text {
  display: block;
  width: 80%;
  margin: 5px auto;
  padding: 5px 10px;
  height: 2.5rem;
  min-width: 125px;
  border: 1px solid black;
  border-left: 3px solid;
  border-radius: 5px; 
  transition: border-color .5s ease-out;
}

.review_form_select {
  display: block;
  width: 100%;
  padding: 5px 10px;
  height: 2.5rem;
  border-radius: 5px;
  margin: 5px auto;
  cursor: pointer;
}

legend {
  text-align: center;
  width: 100%;
  margin: 0 auto;
}

.amenities {
  display: flex;
  flex-direction: column;
}

.amenities label {
  display: flex;
  justify-content: space-between;
  margin: 10px, auto;
}

.amenities label > input {
  margin-left: 20px;
  cursor: pointer;
}

fieldset {
  border: none;
}

.review {
  width: 80%;  
  margin: 0 auto;
}

.review label {
  text-align: left;
  margin: 1.5em;
}

textarea { 
  display: block;
  margin: 5px auto;
  padding: 5px 10px;
  width: 100%;
  height: 5rem;
  min-width: 125px;
  border: 1px solid black;
  border-left: 3px solid;
  border-radius: 5px; 
  transition: border-color .5s ease-out;
}


.review_form_controls {
  width: 80%;
  margin: 0 auto;
}

.review_form_controls button {
  display: block;
  min-width: 180px;
  height: 40px;
  margin: 20px auto;
  border-radius: 5px;
  cursor: pointer;
  background-color: transparent;
  font-size: 1.1em;
}

.cancel {
  border-color: #ED254E;
}

.submit {
  border-color: #064789;
}

@media screen and (min-width: 450px) {
  .review_form {
    width: 75%;
    margin: auto;
  }

  .review_form_controls {
    display: flex;
    justify-content: center;
  }

  .review_form_controls button {
    display: inline-block;
    width: 20px;
  }

}