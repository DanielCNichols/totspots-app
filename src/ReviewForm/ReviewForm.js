import React from 'react'

export default function ReviewForm() {

  return (
    <section class="main">
      <header>
        <h2>Add A Review</h2>
      </header>

      <h3>Bull McCabe's</h3>
      <form action="">
        <fieldset>
          <legend>
            <label for="Price">Price
              <select name="Price" name="price" id="price">
                <option value="">Please select</option>
                <option value="1">$</option>
                <option value="2">$$</option>
                <option value="3">$$$</option>
                <option value="4">$$$$</option>
                <option value="5">$$$$$</option>
              </select>
            </label>
            <label for="Volume">Volume Level
              <select name="Volume" id="Volume">
                <option value="">Please select</option>
                <option value="1">Library</option>
                <option value="2">Coffee Shop</option>
                <option value="3">Restaurant</option>
                <option value="4">Bar/Brewery</option>
                <option value="5">Concert</option>
              </select>
            </label>
            <label for="Rating"
              >Overall Rating
              <select>
                <option value="">Please select</option>
                <option value="1">&#x2605;
                  </option>
                <option value="2">&#x2605;
                    &#x2605;
                  </option>
                <option value="3">&#x2605;
                    &#x2605;
                    &#x2605;
                  </option>
                <option value="4">&#x2605;
                    &#x2605;
                    &#x2605;
                    &#x2605;
                  </option>
                <option value="5">&#x2605;
                    &#x2605;
                    &#x2605;
                    &#x2605;
                    &#x2605;
                  </option>
              </select>
            </label>
          </legend>
          </fieldset>
          <fieldset>
            <legend>Amenities
            <div class="amenities">
              <label for="StrollerAccessible">Stroller Accessible
                <input
                  type="checkbox"
                  name="StrollerAccessible"
                  value="StrollerAccessible"
                />
              </label>
              <label for="PlayArea">Play Area
                <input type="checkbox" name="PlayArea" value="PlayArea" />
              </label>
              <label for="Changingtable"> Changing Table
                <input
                  type="checkbox"
                  name="Changingtable"
                  value="Changingtable"
                />
                </label>
                <label for="Dogs">Dogs Welcome
                <input type="checkbox" name="DogsWelcome" value="Dogs" />
              </label>
              <label for="Fastcheckout"> Fast Checkout
                <input type="checkbox" name="Fastcheckout" value="Fast" />
              </label>
              <label for="KidsNight"> Kids Night Deals
                <input type="checkbox" name="KidsNight" value="KidsNight" />
              </label>
              <label for="OutdoorSeating"> Outdoor Seating Available
                <input type="checkbox" name="OutdoorSeating" value="Outdoor" />
              </label>
            </div>  
            </legend>
          </fieldset>
          <div class="review">
            <label for="review">Write your review
              <input type="text" name="review" placeholder="Tell us about your visit"/>
            </label>
          </div>
        <button type="submit">Submit</button>
        <button>Cancel</button>
      </form>
    </section>
  )
}