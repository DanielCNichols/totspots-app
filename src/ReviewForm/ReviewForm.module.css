* {
  box-sizing: border-box;
}

.form {
  padding-bottom: 100px;
  /* overflow-y: auto; */
}
.controls {
  width: 100%;
  height: 3em;
  display: flex;
  justify-content: space-between;
  /* align-items: center; */
  border-bottom: 2px solid black;
  padding-bottom: 5px;
  margin-bottom: 10px;
}

.formElement,
.reviewField {
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  padding: 0 5px;
}

.formElement input,
.reviewField textarea {
  background-color: transparent;
  border-bottom: 2px solid black;
  border-top: none;
  border-left: none;
  border-right: none;
  padding: 5px;
}

.formElement input:focus,
.reviewField textarea:focus {
  outline: none;
}

.formElement label,
.reviewField label {
  text-align: left;
}

.reviewField {
  height: 100%;
}
.reviewField textarea {
  /* border: 1px solid black; */
  height: 100%;
}
.review {
  width: 95%;
  height: 200px;
  text-align: left;
  margin: auto;
  padding: 10px;
  border: 1px solid black;
}
.review:focus {
  outline: none;
}

/* * {
  box-sizing: border-box;
}
.reviewForm {
  height: 100%;
  display: flex;
  flex-direction: column;
  margin-top: 1.2em;
}

.reviewForm fieldset {
  border: none;
}

.reviewForm legend {
  font-size: 1.2em;
  font-weight: bold;
  text-align: center;
  margin-bottom: 0.8em;
}

.formElement {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.rating {
  display: flex;
  flex-direction: column;
  grid-template-columns: repeat(2, auto);
  column-gap: 10px;
}

.formElement {
  text-align: center;
}

.formElement label {
  font-weight: bold;
  text-align: center;
  margin: 0 auto 10px auto;
}

.rating select {
  width: 60%;
  height: 35px;
  padding: 5px;
  margin: 1em auto;
}

.amenitiesContainer {
  display: grid;
  grid-template-columns: repeat(2, auto);
  width: 100%;
  margin: 0 auto;
}

.amenityElement {
  display: flex;
  justify-content: left;
  align-items: center;
  margin: 10px 0;
  word-wrap: break-word;
  width: 100%;
}

.amenitiesContainer label {
  margin-left: 10px;
  word-wrap: break-word;
  width: 70%;
}

.formElement textarea {
  width: 80%;
  height: 60px;
  margin: 10px auto;
}

.reviewFormControls {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  column-gap: 10px;
  width: 100%;
  padding: 5px;
}

.reviewFormControls button {
  height: 2.5em;
} */
