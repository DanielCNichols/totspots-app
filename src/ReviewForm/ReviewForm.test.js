import React from 'react';
import ReactDOM from 'react-dom'
import ReviewForm from './ReviewForm.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'



it('renders ReviewForm without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <ReviewForm />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders ReviewForm as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <ReviewForm name="ReviewForm"/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
