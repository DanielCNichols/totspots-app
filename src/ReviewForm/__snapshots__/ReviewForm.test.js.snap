// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ReviewForm as expected 1`] = `
<section
  className="review_form_component"
>
  <header>
    <h2>
      Add a review
    </h2>
  </header>
  <h3 />
  <form
    className="review_form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Your Review
        <div
          className="review_form_selects"
        >
          <label
            htmlFor="price"
          >
            Price
            <select
              aria-required="true"
              className="review_form_select"
              id="price"
              name="price"
              required={true}
            >
              <option
                value=""
              >
                Please select
              </option>
              <option
                value={1}
              >
                $
              </option>
              <option
                value={2}
              >
                $$
              </option>
              <option
                value={3}
              >
                $$$
              </option>
              <option
                value={4}
              >
                $$$$
              </option>
              <option
                value={5}
              >
                $$$$$
              </option>
            </select>
          </label>
          <label
            htmlFor="volume"
          >
            Describe the volume level
            <select
              aria-required="true"
              className="review_form_select"
              id="volume"
              name="volume"
              required={true}
            >
              <option
                value=""
              >
                Please select
              </option>
              <option
                value={1}
              >
                ðŸ“£
              </option>
              <option
                value={2}
              >
                ðŸ“£ðŸ“£
              </option>
              <option
                value={3}
              >
                ðŸ“£ðŸ“£ðŸ“£
              </option>
              <option
                value={4}
              >
                ðŸ“£ðŸ“£ðŸ“£ðŸ“£
              </option>
              <option
                value={5}
              >
                ðŸ“£ðŸ“£ðŸ“£ðŸ“£ðŸ“£
              </option>
            </select>
          </label>
          <label
            htmlFor="rating"
          >
            Overall rating
            <select
              aria-required="true"
              className="review_form_select"
              id="rating"
              name="rating"
              required={true}
            >
              <option
                value=""
              >
                Please select
              </option>
              <option
                value={1}
              >
                â˜…
              </option>
              <option
                value={2}
              >
                â˜…â˜…
              </option>
              <option
                value={3}
              >
                â˜…â˜…â˜…
              </option>
              <option
                value={4}
              >
                â˜…â˜…â˜…â˜…
              </option>
              <option
                value={5}
              >
                â˜…â˜…â˜…â˜…â˜…
              </option>
            </select>
          </label>
        </div>
      </legend>
    </fieldset>
    <fieldset>
      <legend>
         
        Features
        <div
          className="review_amenities"
        >
          <div
            className="left"
          >
            <label
              htmlFor="StrollerAccessible"
            >
              Stroller Accessible
              <input
                aria-label="Stroller Accessible"
                className="checkBox"
                name="amenities"
                onChange={[Function]}
                type="checkbox"
                value="1"
              />
            </label>
            <label
              htmlFor="PlayArea"
            >
              Play Area
              <input
                aria-label="Play area"
                className="checkBox"
                name="PlayArea"
                onChange={[Function]}
                type="checkbox"
                value="2"
              />
            </label>
            <label
              htmlFor="Changingtable"
            >
               
              Changing Table
              <input
                aria-label="Changing table"
                className="checkBox"
                name="amenities"
                onChange={[Function]}
                type="checkbox"
                value="3"
              />
            </label>
          </div>
          <div
            className="right"
          >
            <label
              htmlFor="Dogs"
            >
              Dogs Welcome
              <input
                aria-label="Dogs Welcome"
                className="checkBox"
                name="amenities"
                onChange={[Function]}
                type="checkbox"
                value="4"
              />
            </label>
            <label
              htmlFor="Fastcheckout"
            >
               
              Fast Checkout
              <input
                aria-label="Fast Checkout"
                className="checkBox"
                name="amenities"
                onChange={[Function]}
                type="checkbox"
                value="5"
              />
            </label>
            <label
              htmlFor="KidsNight"
            >
               
              Kids Night Deals
              <input
                aria-label="kids night deals"
                className="checkBox"
                name="amenities"
                onChange={[Function]}
                type="checkbox"
                value="6"
              />
            </label>
            <label
              htmlFor="OutdoorSeating"
            >
               
              Outdoor Seating Available
              <input
                aria-label="outdoor seating"
                className="checkBox"
                name="amenities"
                onChange={[Function]}
                type="checkbox"
                value="7"
              />
            </label>
          </div>
        </div>
      </legend>
    </fieldset>
    <div
      className="review_form"
    >
      <label
        htmlFor="review"
      >
        Tell us about your visit
        <textarea
          aria-label="tell us about your visit"
          aria-required="true"
          className="review_form_textarea"
          name="content"
          placeholder="Tell us about your visit"
          type="text"
        />
      </label>
    </div>
    <div
      className="review_form_controls"
    >
      <button
        className="cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="submit"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</section>
`;
